tests/test_controller.py

```diff
@@ -170,6 +173,28 @@ def test_authenticated_patron_correct_credentials(self):
         value = self.controller.authenticated_patron(dict(username="5", password="5555"))
         assert isinstance(value, Patron)
 
+
+    def test_authentication_sends_proper_headers(self):
+        '''
+        Make sure the reals header has quotes around the realm name.  
+        Without quotes, some iOS versions don't recognize the header value.
+        '''
+        response = self.controller.authenticate()
+        
+        found_realm = False
+        for header_key, header_value in response.headers:
```

Comment:
You should also be able to look up the header directly in response.headers without iterating over all of them.